src/main/scala/example/CborAkkaSerializer.scala

```diff
@@ -0,0 +1,39 @@
+package example
+
+import akka.serialization.Serializer
+import io.bullet.borer.{Cbor, Codec, Decoder, Encoder}
+
+import scala.reflect.ClassTag
+
+trait CborAkkaSerializer[Ser] extends Serializer {
```

Comment:
Whoops it looks like Akka (2.6.13) requires the serializer to be a class and provide a specific c'tor:

```
[info] foo.DateTimeJdbcTypeIntegrationSpec *** ABORTED *** (2 milliseconds)
[info]   java.lang.NoSuchMethodException: example.CborAkkaSerializer.<init>(akka.actor.ExtendedActorSystem, java.lang.String)
[info]   at java.base/java.lang.Class.getConstructor0(Class.java:3349)
[info]   at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2553)
[info]   at akka.actor.ReflectiveDynamicAccess.$anonfun$createInstanceFor$1(ReflectiveDynamicAccess.scala:38)
[info]   at scala.util.Try$.apply(Try.scala:213)
[info]   at akka.actor.ReflectiveDynamicAccess.createInstanceFor(ReflectiveDynamicAccess.scala:35)
[info]   at akka.actor.ReflectiveDynamicAccess.$anonfun$createInstanceFor$5(ReflectiveDynamicAccess.scala:48)
[info]   at scala.util.Success.flatMap(Try.scala:251)
[info]   at akka.actor.ReflectiveDynamicAccess.createInstanceFor(ReflectiveDynamicAccess.scala:47)
[info]   at akka.serialization.Serialization$$anonfun$serializerOf$1$$anonfun$applyOrElse$1.applyOrElse(Serialization.scala:396)
[info]   at akka.serialization.Serialization$$anonfun$serializerOf$1$$anonfun$applyOrElse$1.applyOrElse(Serialization.scala:392)
[info]   at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
[info]   at scala.util.Failure.recoverWith(Try.scala:236)
[info]   at akka.serialization.Serialization$$anonfun$serializerOf$1.applyOrElse(Serialization.scala:392)
[info]   at akka.serialization.Serialization$$anonfun$serializerOf$1.applyOrElse(Serialization.scala:390)
[info]   at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
[info]   at scala.util.Failure.recoverWith(Try.scala:236)
[info]   at akka.serialization.Serialization.serializerOf(Serialization.scala:390)
[info]   at akka.serialization.Serialization.$anonfun$serializers$2(Serialization.scala:424)
[info]   at scala.collection.TraversableLike$WithFilter.$anonfun$map$2(TraversableLike.scala:874)
[info]   at scala.collection.immutable.HashMap$HashMap1.foreach(HashMap.scala:394)
[info]   at scala.collection.immutable.HashMap$HashTrieMap.foreach(HashMap.scala:721)
[info]   at scala.collection.immutable.HashMap$HashTrieMap.foreach(HashMap.scala:721)
[info]   at scala.collection.TraversableLike$WithFilter.map(TraversableLike.scala:873)
[info]   at akka.serialization.Serialization.<init>(Serialization.scala:424)
[info]   at akka.serialization.SerializationExtension$.createExtension(SerializationExtension.scala:18)
[info]   at akka.serialization.SerializationExtension$.createExtension(SerializationExtension.scala:14)
[info]   at akka.actor.ActorSystemImpl.registerExtension(ActorSystem.scala:1158)
[info]   at akka.actor.ActorSystemImpl.$anonfun$loadExtensions$1(ActorSystem.scala:1201)
[info]   at scala.collection.Iterator.foreach(Iterator.scala:943)
[info]   at scala.collection.Iterator.foreach$(Iterator.scala:943)
[info]   at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[info]   at scala.collection.IterableLike.foreach(IterableLike.scala:74)
[info]   at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
[info]   at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
[info]   at akka.actor.ActorSystemImpl.loadExtensions$1(ActorSystem.scala:1195)
[info]   at akka.actor.ActorSystemImpl.loadExtensions(ActorSystem.scala:1214)
[info]   at akka.actor.ActorSystemImpl.liftedTree2$1(ActorSystem.scala:1035)
[info]   at akka.actor.ActorSystemImpl._start$lzycompute(ActorSystem.scala:1022)
[info]   at akka.actor.ActorSystemImpl._start(ActorSystem.scala:1022)
[info]   at akka.actor.ActorSystemImpl.start(ActorSystem.scala:1045)
[info]   at akka.actor.typed.ActorSystem$.createInternal(ActorSystem.scala:290)
[info]   at akka.actor.typed.ActorSystem$.apply(ActorSystem.scala:204)
[info]   at akka.actor.testkit.typed.scaladsl.ActorTestKit$.apply(ActorTestKit.scala:108)
[info]   at foo.utils.IntegrationSpec.$init$(IntegrationSpec.scala:35)
[info]   at foo.DateTimeJdbcTypeIntegrationSpec.<init>(DateTimeJdbcTypeIntegrationSpec.scala:23)
[info]   at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[info]   at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[info]   at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[info]   at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
[info]   at java.base/java.lang.Class.newInstance(Class.java:584)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:450)
[info]   at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:413)
[info]   at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[info]   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[info]   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[info]   at java.base/java.lang.Thread.run(Thread.java:834)
```

Reply:
Note that `akka.serialization.jackson.JacksonCborSerializer` has this exact c'tor defined indeed:

```scala

/**
 * INTERNAL API: only public by configuration
 *
 * Akka serializer for Jackson with CBOR.
 */
@InternalApi private[akka] final class JacksonCborSerializer(system: ExtendedActorSystem, bindingName: String)
    extends JacksonSerializer(
      system,
      bindingName,
      JacksonObjectMapperProvider(system).getOrCreate(bindingName, Some(new CBORFactory)))
```

Reply:
We couldn't reproduce that problem. If this c'tor is required, it needs to be in the generated `BorerAkkaSerializer` class, which is not yet included in the library

Reply:
Okay, my misunderstanding, I've tried to use `example.CborAkkaSerializer` as a serializer directly instead of creating my own class like `TestBorerAkkaSerializer` 🤦🏻